spec/async/http/cache/general_spec.rb

```diff
@@ -68,43 +68,113 @@
 				else
 					Protocol::HTTP::Response[200, [['cache-control', 'max-age=1, public'], ['vary', 'user-agent']], ['Hello', ' ', 'World']]
 				end
-				
+
 				if request.head?
 					response.body = Protocol::HTTP::Body::Head.for(response.body)
 				end
-				
+
 				response
 			end
 		end
-		
+
 		let(:user_agents) {[
 			'test-a',
 			'test-b',
 		]}
-		
+
 		it "should cache GET requests" do
 			2.times do
 				user_agents.each do |user_agent|
 					response = subject.get("/", {'user-agent' => user_agent})
 					expect(response.headers['vary']).to include('user-agent')
-					expect(response.read).to be == user_agent
+					expect(response.read).to eq(user_agent)
 				end
 			end
-			
-			expect(store.index.size).to be 2
+
+			expect(store.index.size).to eq(2)
 		end
-	end
-	
+  end
+
+  context 'cache writes' do
+    context 'by response code' do
+      let(:app) do
+        Protocol::HTTP::Middleware.for do |_request|
+          Protocol::HTTP::Response[response_code, []] # no response body?
```

Comment:
I would like to include a response body because that's typical. But I couldn't get the tests to pass, due to Async waiting and ultimately timing out:
```
Waiting for Async::HTTP::Protocol::HTTP11 pool to drain: #<Async::Pool::Controller(2/∞) 1/1/1;1/1/1>
```
I could use some help with that.

Reply:
I'll take a look.

Reply:
This should be fixed now.